### ISSUE TITLE
Consider using different prop nams in RootContainer

### ISSUE THREAD

**Issue by Kureev** [facebook/relay#785]
Hi there,

I'm pretty new to Relay, so correct me if I'm wrong, but it seems that `Relay.RootContainer` doesn't actually support vanilla `Components`, but a `Relay.Containers` instead.

Also, `route` is a pretty confusing name, does it make any sense to change it to something more suitable like `query`?

So, the idea is to use change a public interface to something like this:
```jsx
<Relay.RootContainer
  ...
  Container={SomeRelayContainer}
  query={SomeRelayQuery} />
```

Does it makes any sense?

**Comment by wincent** [comment]
This is good feedback @Kureev. Thanks.

We are indeed moving away from the "route" terminology (see for example #456 and #503). If you look at `RelayRenderer`, which is a newer abstraction that `Relay.RootContainer`, you'll see that we've already moved towards using `RelayQueryConfig` shapes (passed in via a `queryConfig` prop) instead of routes, and we are shooting for `Container` rather than `Component` too (our inline docs actually say `Container`, but the implementation uses `Component`, so either the docs or the code are wrong; I'll see if I can fix this).

In the interests of not creating unnecessary compatibility-breaking changes, I think we're unlikely to make any major changes to the `RootContainer` API, so I am going to close this one for now, and I'll loop back with a link once I've looked into the `Container` vs `Component` issue on `RelayRenderer`.

Thanks once again for your input.
